tests/test_chunkers.py

```diff
@@ -0,0 +1,168 @@
+        "filepaths": document_paths,
+        "documents": [read_document_contents(d) for d in document_paths],
+    }]
+
+
+@pytest.fixture
+def documents_dir():
+    return Path(__file__) / "testdata" / "sample_documents"
+
+
+@pytest.mark.parametrize(
+    "filepaths, chunker_type",
+    [
+        ([Path("document.md")], TextSplitChunker),
+        ([Path("document.pdf")], ContextAwareChunker),
+    ],
+)
+def test_chunker_factory(filepaths, chunker_type):
+    """Test that the DocumentChunker factory class returns the proper Chunker type"""
+    leaf_node = [{
+        "documents": "Lorem ipsum",
+        "taxonomy_path": Path(),
+        "filepaths": filepaths,
+    }]
+    chunker = DocumentChunker(leaf_node=leaf_node)
+    assert isinstance(chunker, chunker_type)
+
+
+def test_split_docs_by_filetype(filepaths):
+    documents = ["Lorem", "ipsum"]
+    md_doc_path = Path("document.md")
+    pdf_doc_path = Path("document.pdf")
+
+    doc_dict = DocumentChunker._split_docs_by_filetype()
+    # TODO
+
+    with pytest.raises(ValueError("Received multiple document types")):
+        md_chunker = DocumentChunker(leaf_node=invalid_leaf_node)
+
+
+class TestTextSplitChunker():
+    @pytest.fixture
+    def chunker():
+        pass
+
+    def test_chunk_documents():
+        pass
+
+    def test_chunk_docs_wc_exceeds_ctx_window(self):
+        with pytest.raises(ValueError) as exc:
+            chunking.chunk_document(
+                documents=testdata.documents,
+                chunk_word_count=1000,
+                server_ctx_size=1034,
+            )
+        assert (
+            "Given word count (1000) per doc will exceed the server context window size (1034)"
+            in str(exc.value)
+        )
+
+    def test_chunk_docs_long_lines(self):
+        # TODO see if this is applicable to context-aware
+        chunk_words = 50
+        chunks = chunking.chunk_document(
+            documents=testdata.long_line_documents,
+            chunk_word_count=chunk_words,
+            server_ctx_size=4096,
+        )
+        max_tokens = chunking._num_tokens_from_words(chunk_words)
+        max_chars = chunking._num_chars_from_tokens(max_tokens)
+        max_chars += chunking._DEFAULT_CHUNK_OVERLAP  # add in the chunk overlap
+        max_chars += 50  # and a bit extra for some really long words
+        for chunk in chunks:
+            assert len(chunk) <= max_chars
+
+    def test_chunk_docs_chunk_overlap_error(self):
+        # TODO check if applicable to context-aware
+        with pytest.raises(ValueError) as exc:
+            chunking.chunk_document(
+                documents=testdata.documents,
+                chunk_word_count=5,
+                server_ctx_size=1034,
+            )
+        assert (
+            "Got a larger chunk overlap (100) than chunk size (24), should be smaller"
+            in str(exc.value)
+        )
+
+
+class TestContextAwareChunker():
+    @pytest.fixture
+    def chunker(documents_dir):
+
```

Comment:
Attempting to run the unit tests from this PR locally (via `tox -e py3-unit`), I'm getting `IndentationError: expected an indented block after function definition on line 128` from this line because there's nothing in the function body - ie needs something here for the code to actually run.